### PR TITLE
fix resource exhaustion bug #190

### PR DESCRIPTION
**Pull request by mofeing**
Disables multithreaded MLIR contexts as suggested in https://github.com/jax-ml/jax/issues/16272

the test needs to run `@compile` at least a number somewhat above 8000 times in macOS (I put 10,000), so it increases core test time from 2 minutes to... 7 minutes 😅

### CODE REVIEW

File: src/Compiler.jl [hunk 1]
```diff
@@ -254,6 +254,7 @@ end
 
 function compile_mlir(f, args; kwargs...)
     ctx = MLIR.IR.Context()
```
Comment (wsmoses) [inline src/Compiler.jl:256]: the context constructor takes an optional argument to disable mutlithreadedd, we shouldn't call it in a follow up method

Comment (mofeing) [reply src/Compiler.jl:256]: do you mean `mlirContextCreateWithThreading` and `mlirContextCreateWithRegistry`? `MLIR.IR.Context` doesn't use them currently

Comment (wsmoses) [reply src/Compiler.jl:256]: yeah we should call mlirContextCreateWithThreading(false)

Comment (wsmoses) [reply src/Compiler.jl:256]: ideally actually yeah the registry and threading constructor

Comment (mofeing) [reply src/Compiler.jl]: done

**Comment by mofeing** [comment]
@wsmoses what do we do about the test time problem? should I remove the test?

**Review by wsmoses — APPROVED** [review]

**Comment by wsmoses** [comment]
@mofeing "DialectRegistry not defined"

**Comment by wsmoses** [comment]
> @wsmoses what do we do about the test time problem? should I remove the test?

Honestly I'm fine either way. I'm okay not testing this as it's pretty fundamental for setup.

File: test/compile.jl [hunk 2]
```diff
@@ -43,4 +43,17 @@ Base.sum(x::NamedTuple{(:a,),Tuple{T}}) where {T<:Reactant.TracedRArray} = (; a=
...
+        y = Reactant.to_rarray(x)
+        @test try
+            for _ in 1:10_000
+                f = @compile sum(y)
+            end
+            true
+        catch e
+            false
+        end
+    end
```
Comment (mofeing) [inline test/compile.jl:58]:
```suggestion

    # disabled due to long test time (core tests go from 2m to 7m just with this test)
    # @testset "resource exhaustation bug (#190)" begin
    #     x = rand(2, 2)
    #     y = Reactant.to_rarray(x)
    #     @test try
    #         for _ in 1:10_000
    #             f = @compile sum(y)
    #         end
    #         true
    #     catch e
    #         false
    #     end
    # end
```

**Pull request merged**